### DISCUSSION TITLE
Resetting forces

### DISCUSSION THREAD

**Discussion by niansa** [jrouwe/JoltPhysics#243] [Q&A] [answered]
Hey!

Very nice project, currenty porting this into the [rbfx](https://github.com/rbfx/rbfx) game engine.
However I don't know how to reset the forces applied to a `JPH::Body`. Can you give me a hint?

Thanks!
niansa

**Comment by jrouwe** [comment]
Hello,

I'm assume we're talking about Body::AddForce? Forces are reset every physics update, so you don't need to reset them (you just need to keep applying them every frame).

Regards,
Jorrit.

**Comment by niansa** [comment]
Oh, what I actually mean is the effect of the force. Even tho the effect is applied just once, between two steps, its effect will stay until it faded out.
And that effect is what I want to stop early.

**Comment by jrouwe** [comment, marked as answer]
The 'effect' of a force is a change in velocity (and thereby also position) in the next update. There is no effect beyond that, so there's nothing to 'fade out'. Once moving, a body will continue to move forever at a constant velocity until an opposing force is applied, usually through friction or damping (see: BodyCreationSettings::mFriction, mLinearDamping and mAngularDamping), but I wouldn't call this 'resetting' or 'fading out' of force.

**Comment by niansa** [comment]
Oh, wow, I actually didn't expect it to be that simple. Thanks for the explaination! I somehow expected it to be rocket science or something. :grin:
